plugins/out_s3/s3.c

```diff
@@ -2132,11 +2144,35 @@ static void cb_s3_flush(struct flb_event_chunk *event_chunk,
                                     event_chunk->tag,
                                     flb_sds_len(event_chunk->tag));
 
+    if (upload_file == NULL) {
+        /* unpack msgpack */
+        msgpack_unpacked_init(&result);
+
+        /* Get the first record timestamp */
+        while (msgpack_unpack_next(&result,
+                                   event_chunk->data,
+                                   event_chunk->size, &off) == MSGPACK_UNPACK_SUCCESS) {
+            flb_time_pop_from_msgpack(&tms, &result, &obj);
+            if (&tms.tm.tv_sec != 0) {
+                file_first_log_time = tms.tm.tv_sec;
+                break;
+            }
+        }
+
+        msgpack_unpacked_destroy(&result);
+    }
+
+    /* Get file_first_log_time from upload_file */
+    if (!file_first_log_time) {
```

Comment:
now that its not a pointer, you can't check it like this. Instead, I think we can initialize it to zero and then check if its still  zero here.